src/Microsoft.Diagnostics.EventFlow.Outputs.ApplicationInsights/Configuration/ApplicationInsightsOutputConfiguration.cs

```diff
@@ -10,6 +10,20 @@ namespace Microsoft.Diagnostics.EventFlow.Configuration
     public class ApplicationInsightsOutputConfiguration: ItemConfiguration
     {
         public string InstrumentationKey { get; set; }
+        public string ConnectionString { get; set; }
         public string ConfigurationFilePath { get; set; }
+
+        public bool Validate(out string validationError)
+        {
+            validationError = null;
+
+            if (string.IsNullOrWhiteSpace(this.InstrumentationKey) && string.IsNullOrWhiteSpace(ConnectionString) && string.IsNullOrWhiteSpace(ConfigurationFilePath))
```

Comment:
Good question. If the iKey is not there, an error will be reported via health reporter, but the output will finish initialization. Nothing will happen (no-op) when any events reach the output. 

This might be a bit cumbersome in some environments. Service Fabric in particular, if `ServiceFabricHealthReporter` is used--reporting an error will make the instance "unhealthy".  I will change the condition to a warning.